lambda/src/google.js

```diff
@@ -0,0 +1,80 @@
+import AWS from 'aws-sdk'
+
+let fs = require('fs');
+let GoogleSpreadsheet = require('google-spreadsheet');
+let s3 = new AWS.S3();
+
+export function getSheetsJson(file, sheetId){
+  const keysFileName = file;
+  const doc = new GoogleSpreadsheet(sheetId);
+  const keysFilePath = "/tmp/" + keysFileName;
+
+  return new Promise((resolve, reject) =>{
+    downloadGoogleCreds()
+      .then(saveCredsFile)
+      .then(configureGoogleSdkKeys)
+      .then(getSheetsRowsJson)
+      .then(function(rows) {
+        return resolve(rows);
+      })
+    });
+
+    function downloadGoogleCreds() {
+      const params = { Bucket: "guconf-flexible/editorial-tools-help", Key: keysFileName };
+      console.log("In creds function");
+      return new Promise((resolve, reject) =>{
+        s3.getObject(params, function(err, response) {
+          if (err) {
+            console.error("failed to retreive google credentials", err);
+            reject(err);
+            return;
+          }
+          return resolve(response.Body.toString());
+        });
+      });
+    }
+
+    function saveCredsFile(response) {
+      console.log("saving creds file");
+      return new Promise((resolve, reject) =>{
+        fs.writeFile(keysFilePath, response, function(err) {
+          if (err) {
+            console.error("could not save keys file", err)
+            reject(err);
+            return;
+          }
+          return resolve();
+        });
+      });
+    }
+
+    function configureGoogleSdkKeys() {
+      console.log("applying google keys");
+      const creds = require(keysFilePath);
```

Comment:
It's the weirdness of the google sdk, it wants to read from a file, and won't do it any other way strangely

Reply:
🤔 This is probably because the `useServiceAccountAuth` expects either a JS object or the required path - you can maybe `resolve` with `JSON.parse(response.Body)` to pass a JS object? If not then :man_shrugging: